pkg/sfu/connectionquality/connectionstats.go

```diff
@@ -51,14 +50,60 @@ type ConnectionStats struct {
 func NewConnectionStats(params ConnectionStatsParams) *ConnectionStats {
 	return &ConnectionStats{
 		params:           params,
-		trackSource:      livekit.TrackSource_UNKNOWN,
-		score:            5.0,
+		codecName:        getCodecNameFromMime(params.MimeType), // LK-TODO: have to notify on codec change
+		normFactors:      [buffer.DefaultMaxLayerSpatial + 1]float32{1, 1, 1},
+		score:            MaxScore,
 		maxExpectedLayer: buffer.InvalidLayerSpatial,
 		done:             make(chan struct{}),
 	}
 }
 
-func (cs *ConnectionStats) Start() {
+func (cs *ConnectionStats) Start(trackInfo *livekit.TrackInfo) {
+	cs.lock.Lock()
+	cs.trackInfo = trackInfo
+
+	//
+	// get best score and calculate normalization factor.
+	// Raitonale: MOS modeling will yield a max score for a specific codec.
+	// That is outside the control of an SFU. SFU can impair quality due
+	// network issues. So, get the maximum for given codec and normalize
+	// it to the highest score. So, under perfect conditions, it will
+	// yield a MOS of 5 which is the highest rating. Any SFU induced impairment
+	// or network impairment will result in a lower score and that is
+	// the quality that is under SFU/infrastructure control.
+	//
+	if trackInfo.Type == livekit.TrackType_AUDIO {
+		// LK-TODO: would be good to have audio expected bitrate in Trackinfo
+		params := TrackScoreParams{
+			Duration: time.Second,
+			Codec:    cs.codecName,
+			Bytes:    20000 / 8,
+		}
+		cs.normFactors[0] = MaxScore / AudioTrackScore(params, 1)
+	} else {
+		for _, layer := range cs.trackInfo.Layers {
+			spatial := utils.SpatialLayerForQuality(layer.Quality)
+			// LK-TODO: would be good to have expected frame rate in Trackinfo
+			frameRate := uint32(30)
+			switch spatial {
+			case 0:
+				frameRate = 15
```

Comment:
I think this would be difficult to assume, since clients may capture at arbitrary framerate with user-provided tracks.

Agreed we should move this into the protocol itself. I had added it into [this PR](https://github.com/livekit/protocol/pull/120), but have not merged it yet.

Reply:
Yeah, I am using the default settings in our client SDKs for now. Having that in the protocol would be great.